### PR TITLE
Logging improvements

### PR DESCRIPTION
**Pull request by Frede1012**
Added dates to logs, Added method to delete older logs, Added break between scriptdomain runs

As discussed in #103

### CODE REVIEW

**Comment by Frede1012** [comment]
I was also considering changing the Log class to a Logger with two types of constructors. One that takes no variables, and one that takes an `Assembly`. 
If no assembly passed, the logger will use the default assembly and therefor log to `Scripthookdotnet.log` but if you pass an assembly you get `myscript.log`.

Thoughts? @crosire @JohnnyCrazy

**Comment by Frede1012** [comment]
I created a different branch for the new logging implementation. Check it out here: https://github.com/Frede1012/scripthookvdotnet/tree/logging

Thoughts? @crosire @JohnnyCrazy

**Comment by crosire** [comment]
How about doing it the same way settings are handled. A class which takes a log path and a "Script::Log" property to access it inside scripts.

**Comment by Frede1012** [comment]
So the logger takes a path instead of an Assembly? And a logger class is automatically instantiated for each script?

**Comment by crosire** [comment]
Using assemblies would mean we have to change the way settings are handled too, to keep it consistent.

**Comment by Frede1012** [comment]
So the `Logger` class should have two constructors, one that has takes no path and uses the default `Assembly` path (`Scripthookvdotnet.log`) and one that takes a custom path (`myscript.log`)?

**Comment by Frede1012** [comment]
How about this https://github.com/Frede1012/scripthookvdotnet/commit/7219ada1769d46715d006acf9ad640a0a7021c2a ?

**Comment by crosire** [comment]
I was rethinking about the whole thing again. Why do we actually need a logging class for scripts? It's not that hard to write such a thing yourself in a few C# lines ... And it bloats the code.

**Comment by Frede1012** [comment]
That is true, however when we already have code to do it, why not expose it to the scripts?

**Comment by Frede1012** [comment]
You don't agree? @crosire

**Comment by Frede1012** [comment]
Maybe these changes could be merged, and i could create a different pull request for my other implementation for further discussion? @crosire @JohnnyCrazy

**Comment by Frede1012** [comment]
I created a new pull request for the other implementatation (#196), could this be merged? @JohnnyCrazy @crosire

File: source/Log.cpp [hunk 1]
```diff
@@ -5,19 +5,46 @@ namespace GTA
 	using namespace System;
 	using namespace System::Collections::Generic;
 
+	void Log::OnStart()
+	{
+		LogToFile("", false, "##########\n\n\n\n##########");
```
Comment (nevadascout) [inline source/Log.cpp]: Can we change this to remove the newlines? Something like:

```
LogToFile("", false, "########################################");
```

Comment (Frede1012) [inline source/Log.cpp]: Maybe, do you think thats better?

Comment (JohnnyCrazy) [inline source/Log.cpp]: I agree with @nevadascout, 4 new-lines are too much. I would also go with a one-liner

Comment (Frede1012) [inline source/Log.cpp]: Fixed :-)

**Comment by Frede1012** [comment]
?? @crosire @JohnnyCrazy

**Comment by crosire** [comment]
Merged in c7dafc5ed4c9fc7ad123d6d38fba78f5b8786843, though with additional follow up changes in 5cff7ca7c6e03e28a18b888baf0a1899ffe494fe.

**Pull request closed without merging**